cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultPreviewManager.java

```diff
@@ -143,15 +147,20 @@
     this.previewRequestQueue = previewRequestQueue;
     this.previewStore = new DefaultPreviewStore(previewLevelDBTableService);
     this.previewRunnerServiceStopper = previewRunnerServiceStopper;
+    this.messagingService = messagingService;
   }
 
   @Override
-  protected synchronized void startUp() throws Exception {
+  protected void startUp() throws Exception {
     previewInjector = createPreviewInjector();
+    PreviewDataSubscriberService subscriberService = previewInjector.getInstance(PreviewDataSubscriberService.class);
+    subscriberService.startAndWait();
   }
 
   @Override
   protected void shutDown() throws Exception {
+    PreviewDataSubscriberService subscriberService = previewInjector.getInstance(PreviewDataSubscriberService.class);
```

Comment:
Make the `subscriberService` a field and only call the injector.getInstance once in the startUp method. This makes it clear it is the same instance instead of relying on the singleton binding, which is not clear in this context.

Reply:
Good point! fixed.